PWGHF/TableProducer/trackToCollisionAssociator.cxx

```diff
@@ -0,0 +1,148 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file trackToCollisionAssociator.cxx
+/// \brief Associates tracks to collisions considering ambiguities
+///
+/// \author
+
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/ASoAHelpers.h"
+#include "Framework/runDataProcessing.h"
+#include "PWGHF/DataModel/CandidateReconstructionTables.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+using namespace o2::aod;
+
+struct HfTrackToCollisionAssociation {
+  Produces<HfTrackAssoc> association;
+  Produces<HfTrackCompColls> reverseIndices;
+
+  Configurable<float> nSigmaForTimeCompat{"nSigmaForTimeCompat", 3.f, "number of sigmas for time compatibility"};
+
+  using TracksWithSel = soa::Join<Tracks, TracksExtra, TrackSelection>;
+  Filter trackFilter = requireGlobalTrackWoDCAInFilter();
+
+  void processAssocWithTime(Collisions const& collisions,
+                            TracksWithSel const& tracksUnfiltered,
+                            soa::Filtered<TracksWithSel> const& tracks,
+                            BCs const& bcs)
+  {
+    std::vector<int>** collsPerTrack = new std::vector<int>*[tracksUnfiltered.size()];
+    memset(collsPerTrack, 0x0, sizeof(collsPerTrack) * tracksUnfiltered.size());
+
+    // loop over collisions to find time-compatible tracks
+    float nSigmaForTimeCompat2 = nSigmaForTimeCompat * nSigmaForTimeCompat;
+    auto trackBegin = tracks.begin();
+    const auto bOffsetMax = 241;
+    for (const auto& collision : collisions) {
+      float collTime = collision.collisionTime();
+      float collTimeRes2 = collision.collisionTimeRes() * collision.collisionTimeRes();
+      uint64_t collBC = collision.bc().globalBC();
+      bool iteratorMoved = false;
+      for (auto track = trackBegin; track != tracks.end(); ++track) {
+        if (!track.has_collision()) {
+          continue;
+        }
+        int64_t bcOffset = (int64_t)track.collision().bc().globalBC() - (int64_t)collBC;
+        float trackTime{0.};
+        float trackTimeRes2{0.};
+        if (track.isPVContributor()) {
+          trackTime = track.collision().collisionTime(); // if PV contributor, we assume the time to be the one of the collision
+          trackTimeRes2 = 25.f;                          // 1 BC
```

Comment:
I think this is not correct because it should be `25 * 25`

Reply:
Good catch, thanks!